server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -816,10 +824,29 @@ public static RepositoryData snapshotsFromXContent(XContentParser parser, long g
                     indexMetaIdentifiers = parser.mapStrings();
                 }
                 case MIN_VERSION -> {
-                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(), parser);
-                    final Version version = Version.fromString(parser.text());
+                    IndexVersion version;
+                    switch (parser.nextToken()) {
+                        case VALUE_STRING -> {
+                            Version v = Version.fromString(parser.text());
+                            assert v.before(Version.V_8_10_0);
```

Comment:
We could leave out this assertion and just use the same code for parsing the version here and in `parseSnapshots` below I think :) This code is complicated enough as it is (especially its BwC), would be nice to not duplicate that logic.

Reply:
Pulled it out into a shared method